iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.hpp

```diff
@@ -96,7 +96,8 @@ class PortManager
     void deletePortsOfProcess(const RuntimeName_t& runtimeName) noexcept;
 
     const std::atomic<uint64_t>* serviceRegistryChangeCounter() noexcept;
-    runtime::IpcMessage findService(const capro::IdString_t& service, const capro::IdString_t& instance) noexcept;
+    runtime::IpcMessage findService(const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& service,
```

Comment:
An alternative could be to use `cxx::optional<capro::IdString_t>` instead of the `cxx::variant<cxx::Wildcard_t, capro::IdString_t>` construct.

The API would be a bit more understandable, as well as the code and we could get rid of the wildcard struct. But then downside may be that it is not clear for the user that a nullopt is then like a wildcard - on the other hand, what else could it be in the findService context!?

Reply:
Yes, the variant is bad and optional would be better. It would be even better to have overloads with `WildCard_t` in addition but then we would have 4 different combinations. Those can be used if compile time dispatch is possible (and the runtime dispatch can also variant can also call them internally. This gets annoying to extend and maintain though.

Ideally we can remove variant here though, but it is not a priority for internal calls (i.e. not exposed to the user).